File: 547-number-of-provinces/number-of-provinces.cpp
class Solution {
public:

    int visited[200];
    void dfs(int s, vector<vector<int>>& isConnected, int crr) {
        int n = isConnected[0].size();
        
        for(int i=s; i<n; ++i) {
            if(isConnected[crr][i] && !visited[i]) {
                visited[i] = 1;
                dfs(s, isConnected, i);
            } 
        }
    }

    int findCircleNum(vector<vector<int>>& isConnected) {
        int comp = 0;
        for(int i=0; i<200; ++i) {
            visited[i] = 0;
        }
        
        for(int i=0; i<isConnected.size(); ++i) {

            if(visited[i])
                continue;
            
            visited[i] = 1;
            dfs(i+1, isConnected, i);

            ++comp;
        }

        return comp;
    }
};

File: 547-number-of-provinces/number-of-provinces.java
class Solution {

    public int findCircleNum(int[][] isConnected) {
        int V = isConnected.length;
        boolean[] vis = new boolean[V];

        int count = 0;

        for (int i = 0; i < V; i++) {
            if (!vis[i]) {
                count++;
                dfs(i, vis, isConnected);
            }
        }
        return count;
    }

    public void dfs(int node, boolean[] vis, int[][] isConnected) {        
        vis[node] = true;
        for (int i = 0; i < isConnected[node].length; i++) {
            if (!vis[i] && isConnected[node][i] == 1) {
                dfs(i, vis, isConnected);
            }
        }

    }

}